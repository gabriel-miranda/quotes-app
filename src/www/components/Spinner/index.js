import React from 'react';
import PropTypes from 'prop-types';

const SPINNER_STYLE = {
  margin: '0 auto 20px',
};

const Spinner = ({size}) => (
  <div className="spinner spinner-lg is-auth0" style={SPINNER_STYLE}>
    <div className="circle" />
  </div>
);

Spinner.propTypes = {
  size: PropTypes.oneOf([
    'xs',
    'sm',
    'md',
    'lg',
  ]),
};

Spinner.defaultProps = {
  size: 'lg',
};

export default Spinner;
