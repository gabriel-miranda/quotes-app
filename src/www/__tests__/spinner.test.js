/* eslint-env jest */

import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import Spinner from '../components/Spinner';

describe('Spinner test with enzyme', () => {
  it('should show a large spinner if no size prop is set', () => {
    const spinner = shallow(<Spinner />);
    expect(spinner.find('.spinner-lg').length).toEqual(1);
  });

  it('should show an auth0 spinner if no auth0 prop is set', () => {
    const spinner = shallow(<Spinner />);
    expect(spinner.find('.is-auth0').length).toEqual(1);
  });

  it('should show a simple (no auth0) spinner if auth0 prop is set to false', () => {
    const spinner = shallow(<Spinner auth0={false} />);
    expect(spinner.find('.is-auth0').length).toEqual(0);
  });

  it('should show a small spinner if small prop is set', () => {
    const spinner = shallow(<Spinner size="sm" />);
    expect(spinner.find('.spinner-sm').length).toEqual(1);
  });
});

describe('Spinner test with Snapshot Testing', () => {
  it('should show Card correctly', () => {
    const component = renderer.create(<Spinner />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
