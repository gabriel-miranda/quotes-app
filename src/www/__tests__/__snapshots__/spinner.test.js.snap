// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner test with Snapshot Testing should show Card correctly 1`] = `
<div
  className="spinner spinner-lg is-auth0"
  style={
    Object {
      "margin": "0 auto 20px",
    }
  }
>
  <div
    className="circle"
  />
</div>
`;
